sql/expression/function/aggregation/unary_agg_buffers.go

```diff
@@ -125,12 +157,24 @@ func (a *avgBuffer) Update(ctx *sql.Context, row sql.Row) error {
 		return nil
 	}
 
-	v, err = sql.Float64.Convert(v)
-	if err != nil {
-		v = float64(0)
+	switch n := v.(type) {
+	case decimal.Decimal:
+		if sum, ok := a.sum.(decimal.Decimal); ok {
+			a.sum = sum.Add(n)
+		} else {
+			a.sum = decimal.NewFromFloat(a.sum.(float64)).Add(n)
+		}
+	default:
+		val, err := sql.Float64.Convert(n)
+		if err != nil {
+			val = float64(0)
+		}
+		sum, err := sql.Float64.Convert(a.sum)
+		if err != nil {
+			sum = float64(0)
+		}
+		a.sum = sum.(float64) + val.(float64)
```

Comment:
I was expecting this code would need to do the same logic as in `sumBuffer.Update`. Are the differences not important? If we do need that same logic, can we just invoke `sumBuffer.Update` with the row to get the sum updated and not have to repeat the logic?